clang-tools-extra/clangd/InlayHints.cpp

```diff
@@ -368,6 +368,14 @@ static FunctionProtoTypeLoc getPrototypeLoc(Expr *Fn) {
   }
 
   if (auto F = Target.getAs<FunctionProtoTypeLoc>()) {
+    // In some edge cases the AST can contain a "trivial" FunctionProtoTypeLoc
```

Comment:
nit: We have a duplicated one in ` SemaCodeComplete.cpp`, see the `FIXME` on line 337. I guess the same issue  should happen there as well (signaturehelp?). I see options 1) duplicate the workaround 2) unify these two implementations. Either works for me.

Reply:
Thanks for the reminder about this. Now that we've upstreamed `HeuristicResolver`, it's a good place for a unified implementation of this sort of thing. I'll do that in https://github.com/llvm/llvm-project/issues/143240.

Reply:
Posted https://github.com/llvm/llvm-project/pull/143345 for this.